src/utils/stateCollectionEntityUpdate/index.ts

```diff
@@ -0,0 +1,39 @@
+import { Entity } from "../../types/Entity"
+import { StateCollection } from "../../types/StateCollection"
+import { Action } from "../../types/Action"
+import { Update } from "../../types/Update"
+
+/**
+ * `stateCollectionEntityUpdate`
+ * ------------------
+ * Updates `Entity<A>` in `StateCollection<Entity<A>>`
+ **/
+
+export const stateCollectionEntityUpdate = <
+  A extends Entity,
+  State extends StateCollection<A>
+>(
+  state: State,
+) => ({ payload }: Action<Update<A>>) =>
+  state.result.reduce<StateCollection<A>>(
+    (a, b) =>
+      b === payload.id
+        ? {
+            entities: {
+              ...a.entities,
+              [b]: { ...state.entities[b], ...payload.properties },
+            },
+            result: [...a.result, b],
+          }
+        : {
```

Comment:
In this branch of the condition could you just return `a` instead of creating a new object?

Reply:
Sure could and should

Reply:
Hmmm I may be doing this wrong, but it seems that I have to build a new state either way?

Reply:
🤔

Reply:
Would love to be proven wrong 😄

Reply:
@roblafeve If you changed it to something like this:
```
export const stateCollectionEntityUpdate = <
  A extends Entity,
  State extends StateCollection<A>
>(
  state: State,
) => ({ payload }: Action<Update<A>>) =>
  state.result.reduce<StateCollection<A>>(
    (a, b) =>
      b === payload.id
        ? {
            entities: {
              ...a.entities,
              [b]: { ...state.entities[b], ...payload.properties },
            },
            result: a.result,
          }
        : a,
    state,
  )
```

The function will only create a new state when it updates the one entity it needs to, as opposed to having to create a new state for every entity in the collection for every update action.

The main change is starting with the current state instead of starting with an empty state.

Reply:
Yup it was starting with a new state that made this necessary 🤦‍♂ 

You rock 🤘